dms_field_auto_classification/models/dms_classification_template.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class DmsClassificationTemplate(models.Model):
+    _inherit = "dms.classification.template"
+
+    model_id = fields.Many2one(
+        comodel_name="ir.model", string="Model", domain=[("transient", "=", False)]
+    )
+    model = fields.Char(compute="_compute_model", compute_sudo=True)
+    detail_ids = fields.One2many(
+        string="Details",
+        comodel_name="dms.classification.template.detail",
+        inverse_name="parent_id",
+    )
+
+    @api.depends("model_id")
+    def _compute_model(self):
+        for item in self:
+            item.model = item.model_id.model
```

Comment:
I think this will fail with non admin permissions, as models are restricted (and the same for any model access in whole module).

Reply:
This field is compute sudo.
Template must be created by a user with `ir.model` permissions.

I have added to tests a user with permissions in Documents: Manager to confirm that that user can do the whole process (without having permissions to `ir.model`).

Reply:
> Template must be created by a user with `ir.model` permissions.

Only maximum admins have this permission, so this should be avoided (`mis_builder` for example had similar problems).

Reply:
I understand the problem of high permissions, but what suggestion do you propose? 
Another option is not to select a model (`ir.model`) and write the name (res.partner for example) directly.